### PR TITLE
test: add test case about codec

### PR DESCRIPTION
**Pull request by tzssangglass**
1. add test cases about codec 
2. fix bug about encode response body

### CODE REVIEW

**Comment by tzssangglass** [comment]
@dmsolr @spacewander pls review

File: runner-core/src/test/java/org/apache/apisix/plugin/runner/codec/impl/FlatBuffersDecoderTest.java [hunk 1]
```diff
@@ -0,0 +1,127 @@
...
+        A6Request result = flatBuffersDecoder.decode(buffer);
+        Assertions.assertEquals(Code.BAD_REQUEST, ((A6ErrRequest) result).getCode());
+        Assertions.assertTrue(capturedOutput.getOut().contains("receive error data length"));
+    }
+
+    @Test
+    @DisplayName("test get body")
+    void testGetBody() {
+        // {"name":"foo", "value":"bar"}
+        byte[] bytes = new byte[]{1, 0, 0, 64, 12, 0, 0, 0, 0, 0, 6, 0, 8, 0, 4, 0, 6, 0, 0, 0, 4, 0, 0, 0, 1, 0, 0, 0, 12, 0, 0, 0, 8, 0, 12, 0, 8, 0, 4, 0, 8, 0, 0, 0, 8, 0, 0, 0, 12, 0, 0, 0, 3, 0, 0, 0, 98, 97, 114, 0, 3, 0, 0, 0, 102, 111, 111, 0};
```
Comment (dmsolr) [inline runner-core/src/test/java/org/apache/apisix/plugin/runner/codec/impl/FlatBuffersDecoderTest.java:108]: Using "{\"name\":\"foo\", \"value\":\"bar\"}".getBytes() directly, is more readable and reasonable.

File: runner-core/src/test/java/org/apache/apisix/plugin/runner/codec/impl/FlatBuffersEncoderTest.java [hunk 2]
```diff
@@ -0,0 +1,172 @@
...
+    @Test
+    @DisplayName("test encode error response(1)")
+    void testErrResponseEncode1() {
+        A6ErrResponse errResponse = new A6ErrResponse(Code.BAD_REQUEST);
+        ByteBuffer result = flatBuffersEncoder.encode(errResponse);
+        byte[] res = new byte[result.remaining()];
+        result.get(res);
+
+        //example
+        byte[] example = new byte[]{0, 0, 0, 12, 8, 0, 0, 0, 4, 0, 4, 0, 4, 0, 0, 0};
```
Comment (dmsolr) [inline runner-core/src/test/java/org/apache/apisix/plugin/runner/codec/impl/FlatBuffersEncoderTest.java:47]: The same as above.

File: runner-core/src/test/java/org/apache/apisix/plugin/runner/codec/impl/FlatBuffersDecoderTest.java (hunk 1 above)
Comment (tzssangglass) [reply runner-core/src/test/java/org/apache/apisix/plugin/runner/codec/impl/FlatBuffersDecoderTest.java]: `bytes` is not equal to `"{"name": "foo", "value": "bar"}".getBytes()`, but consists of a Flatbuffer serialized and assembled with a private data protocol.
I've modified the way the data is generated.

Comment (tzssangglass) [reply runner-core/src/test/java/org/apache/apisix/plugin/runner/codec/impl/FlatBuffersDecoderTest.java]: resolved, pls recheck

**Review by dmsolr — DISMISSED** [review]

**Review by dmsolr — APPROVED** [review]

**Review by spacewander — APPROVED** [review]

**Pull request merged**